File: coletor.py
from bs4 import BeautifulSoup
import requests

def coletor(url, nome):
    r = requests.get(url)
    start = r.text.find('<div class="card shadow-sm resultadoBuscaItem">')
    end = r.text.find('<div class="rodape">')
    recorte = r.text[start:end]
    final = 0
    com_ex = '<a class="bg-light text-dark" style="text-decoration: none;" href="'
    fin_ex = len('" target="_blank">')
    links = [nome]
    c = 0
    while True:
        começo = recorte.find(com_ex) + len(com_ex)
        if começo == 66:
            break
        for k in recorte[começo:]:
            c+= 1
            if k == '>':
                final = começo+c
                links.append(('http://www.imprensaoficial.com.br/' + recorte[começo:final-fin_ex]).replace(' ', ''))
                recorte = recorte[final+1:]
                c = 0
                break
    return links

#a função recebe o nome da pessoa que esta sendo pesquisada apenas por referencia pois a lista retornada 
#tera o primeiro elemento como o nome assim organizando os links junto com o nome procurado

File: coletor teste/coletor.js
import fetch from "node-fetch"

function coletor(link) {
    fetch(link)
.then(res => res.text())
.then(text => {
    //console.log(text)
    let rec_comeco = text.search('<div class="card shadow-sm resultadoBuscaItem">')
    let rec_fim = text.search('<div class="rodape">')
    var recorte = text.substring(rec_comeco, rec_fim)
    let link_comeco = '<a class="bg-light text-dark" style="text-decoration: none;" href="/DO/BuscaDO2001Documento'
    let link_fim = '" target="_blank">'
    var lista_links = []
    while (true) {
        if(recorte.search(link_comeco) == -1){break}
        let comeco = recorte.search(link_comeco) + link_comeco.length + 5
        let pos_comeco = recorte.substr(comeco)
        let link = pos_comeco.substring(0, pos_comeco.search(link_fim))
        lista_links.push('http://www.imprensaoficial.com.br/DO/GatewayPDF' + link)
        let resto = recorte.substring(0, comeco + link.length + link_fim.length)
        recorte = recorte.substr(resto.length)
    }
    return lista_links
})
.catch(err => console.log(err));
}


export default coletor